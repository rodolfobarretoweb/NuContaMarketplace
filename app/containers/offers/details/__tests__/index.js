import React from 'react';
import renderer from 'react-native-test-utils';
import { shallow } from 'enzyme';
import { Details } from '../index';

describe('Details container', () => {
  const props = {
    navigator: {
      setTitle: jest.fn()
    },

    balance: 1000,
    id: 'id',
    price: 10,
    product: {
      name: 'Product name',
      description: 'Description',
      image: 'image.jpg'
    }
  };


  it('renders snapshot correctly', () => {
    const item = renderer(<Details {...props} />);
    expect(item.toJSON()).toMatchSnapshot();
  });

  describe('when i make a purchase', () => {
    const simulate = () => {
      const wrapper = shallow(<Details {...props} />);
      const render = wrapper.dive();

      render.find('TouchableHighlight').simulate('Press');

      return render;
    };

    it('redirect with push method', () => {
      expect(simulate()).toMatchSnapshot();
    });
  });
});
