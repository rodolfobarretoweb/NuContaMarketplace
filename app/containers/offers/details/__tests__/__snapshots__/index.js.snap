// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details container renders snapshot correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "backgroundColor": "#fff",
      "flexGrow": 1,
    }
  }
>
  <View>
    <View>
      <Image
        source={
          Object {
            "uri": "image.jpg",
          }
        }
        style={
          Object {
            "height": 300,
            "width": "100%",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 25,
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        Product name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#782698",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        $10.00
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "bottom": 0,
          "elevation": 3,
          "flex": 1,
          "flexDirection": "row",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "right": 0,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0.1,
            "width": 1,
          },
          "shadowOpacity": 0.3,
        }
      }
    >
      <View
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "#782698",
                "borderRadius": 10,
                "paddingHorizontal": 10,
                "paddingVertical": 8,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "right",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Details container when i make a purchase redirect with push method 1`] = `
<_class
  contentContainerStyle={
    Object {
      "backgroundColor": "#fff",
      "flexGrow": 1,
    }
  }
>
  <View>
    <View>
      <Image
        source={
          Object {
            "uri": "image.jpg",
          }
        }
        style={
          Object {
            "height": 300,
            "width": "100%",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 25,
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        Product name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#782698",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        $10.00
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Description
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "bottom": 0,
          "elevation": 3,
          "flex": 1,
          "flexDirection": "row",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "right": 0,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0.1,
            "width": 1,
          },
          "shadowOpacity": 0.3,
        }
      }
    >
      <View>
        <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "#782698",
              "borderRadius": 10,
              "paddingHorizontal": 10,
              "paddingVertical": 8,
            }
          }
          underlayColor="transparent"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          />
        </TouchableHighlight>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "right",
            }
          }
        />
      </View>
    </View>
  </View>
</_class>
`;
